delivery_postlogistics/models/stock_picking.py

```diff
@@ -0,0 +1,242 @@
+# Copyright 2013-2016 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import base64
+from operator import attrgetter
+
+from odoo import _, api, exceptions, fields, models
+
+from ..postlogistics.web_service import PostlogisticsWebService
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    delivery_fixed_date = fields.Date(
+        "Fixed delivery date", help="Specific delivery date (ZAW3217)"
+    )
+    delivery_place = fields.Char(
+        "Delivery Place", help="For Deposit item service (ZAW3219)"
+    )
+    delivery_phone = fields.Char(
+        "Phone", help="For notify delivery by telephone (ZAW3213)"
+    )
+    delivery_mobile = fields.Char(
+        "Mobile", help="For notify delivery by telephone (ZAW3213)"
+    )
+
+    @api.onchange("carrier_id")
+    def onchange_carrier_id(self):
+        """ Inherit this method in your module """
+        if not self.carrier_id:
+            return
+        # This can look useless as the field carrier_code and
```

Comment:
indeed, it looks useless :)
The related field is readonly... I don't understand the purpose of this.

Reply:
@simahawk value of the related field will not be triggered to update unless the original field is saved to the db. So I think this is a trick to trigger the change of the related field on changing the data of the original field.

The related field decides how the view looks like, so we need it to be updated.

Reply:
If you tell me it works... fine, I learn something new :wink: 
Judging only from code, since `delivery_type` is already related to carrier.delivery_type and is readonly IDT the form is going to trigger an update. But I might be wrong... :man_shrugging:

Reply:
ah I checked again, I am wrong :D. Yes agree that we don't need this function actually.